public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -347,8 +354,16 @@ export class ElasticDatasource {
         const index = result[indexName];
         if (index && index.mappings) {
           const mappings = index.mappings;
-          for (const typeName in mappings) {
-            const properties = mappings[typeName].properties;
+
+          if (configuredEsVersion < 70) {
+            for (const typeName in mappings) {
+              const properties = mappings[typeName].properties;
+              getFieldsRecursively(properties);
+            }
+          } else {
+            // ES 7 finally removed types in indices, and it also removed
+            // them from the API
+            const properties = mappings.properties;
```

Comment:
I think this comment can be removed and instead you can add a test for es7 version. There's an existing test named `When getting fields` that you should be able to copy and change to use es7 version.

Reply:
Ok, here is my attempt at adding some tests for this:  b3b3371 . As far as I can tell, those new tests are working OK. However I get failed tests which _seem_ to be connected: 

```
(node:11169) UnhandledPromiseRejectionWarning: Error: expect(received).toEqual(expected)

Difference:

- Expected
+ Received

- Array [
-   "@timestamp",
-   "beat.name.raw",
-   "beat.name",
-   "beat.hostname",
-   "system.cpu.system",
-   "system.cpu.user",
-   "system.process.cpu.total",
-   "system.process.name",
- ]
+ Array []
(node:11169) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:11169) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:11169) UnhandledPromiseRejectionWarning: Error: expect(received).toEqual(expected)

Difference:

- Expected
+ Received

- Array [
-   "system.cpu.system",
-   "system.cpu.user",
-   "system.process.cpu.total",
- ]
+ Array []
(node:11169) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:11169) UnhandledPromiseRejectionWarning: Error: expect(received).toEqual(expected)

Difference:

- Expected
+ Received

- Array [
-   "@timestamp",
- ]
+ Array []

```

Unfortunately, I'm unfamiliar with the way the test harness works, and can't pinpoint the message above to precise locations in the test suite.

Reply:
There was actually a problem with the tests. ES version wasn't provided in a correct way. Have pushed fix.